idunn/geocoder/nlu_client.py

```diff
@@ -289,10 +296,10 @@ async def get_intentions(self, text, lang, focus=None):
                 )
 
             logger.info("Detected intentions for '%s'", text, extra=logs_extra)
-        else:
-            logger.info("No intention detected for '%s'", text, extra=logs_extra)
-
-        return intentions
+            return [intention]
+        except NluClientException as exp:
+            exp.extra = logs_extra
```

Comment:
`exp.extra` is already set and contains "reason". 
```suggestion
            exp.extra.update(logs_extra)
```